metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -143,22 +145,34 @@ class DebugProvider(
         clazz.setJvmOptions(
           Option(params.jvmOptions).getOrElse(List().asJava)
         )
-        val env =
+
+        val env: List[String] =
           if (params.env != null)
             params.env.asScala.map {
               case (key, value) => s"$key=$value"
             }.toList
           else
             Nil
 
-        clazz.setEnvironmentVariables(env.asJava)
-        Future.successful(
-          new b.DebugSessionParams(
-            singletonList(target.getId()),
-            b.DebugSessionParamsDataKind.SCALA_MAIN_CLASS,
-            clazz.toJson
+        val envFromFile: Future[List[String]] =
+          if (params.envFile != null) {
+            val path = AbsolutePath(params.envFile)(workspace)
+            DotEnvFileParser
+              .parse(path)
+              .map(_.map { case (key, value) => s"$key=$value" }.toList)
+          } else Future.successful(List.empty)
+
+        envFromFile.flatMap { envFromFile =>
```

Comment:
Maybe just `map` and we can skip `Future.successful` below?

Reply:
addressed in 78ec1b1